jdaviz/core/marks.py

```diff
@@ -116,6 +116,9 @@ def set_y_unit(self, unit=None):
                 if ('_pixel_scale_factor' in spec.meta):
                     eqv += _eqv_pixar_sr(spec.meta['_pixel_scale_factor'])
                     y = (self.y * self.yunit).to_value(unit, equivalencies=eqv)
+
+                # for flux <> flux/pix2
+                eqv += _eqv_flux_to_sb_pixel()
```

Comment:
not 100% sure this suggestion is correct, but the logic through both ifs results in undefined value for `y`

```suggestion
                else:
                    # for flux <> flux/pix2
                    eqv += _eqv_flux_to_sb_pixel()
                y = (self.y * self.yunit).to_value(unit, equivalencies=eqv)
```

Reply:
I changed the logic, i think what I have now makes sense for defining 'y'. 

While looking at this snippet, I realized that theres a potentially uncaught issue that might need to be changed in other plugins that do in-plugin unit conversion. Here y is being changed to a new unit, and I added in the new custom equivalency for flux<>flux/pix2 that is combined with the flux<>flux/sr and u.spectral_density equivalencies which enables a lot of conversions however i think there are some unit conversions that won't work in this block. For example, something like ph / (Angstrom s cm2 pix2) and Jy / pix2 would not convert here (even if provided spectral_density, this is part of the 'untranslatable units' issue where the angle needs to be multiplied out first because the equivalencies aren't combining correctly). 

I think every unit.to() in every plugin should pass through utils.flux_conversion rather than using astropy unit translation. I think when I wrote some of this a few weeks ago, I hadn't encountered those cases yet (I was focused on stuff like Jy <> Jy / pix2).

I need to think about this a bit more, and I'll address the rest of the review comments before I investigate that and add more test coverage in plugins for weird unit translations, but I do see a lot of room for unit conversions that would break things that aren't covered by tests (I thought I got them all!)